db/version_set.cc

```diff
@@ -2857,7 +2857,32 @@ Status VersionSet::ProcessManifestWrites(
       last_writer = *(it++);
       assert(last_writer != nullptr);
       assert(last_writer->cfd != nullptr);
-      if (last_writer->cfd != nullptr && last_writer->cfd->IsDropped()) {
+      if (last_writer->cfd->IsDropped()) {
```

Comment:
is it possible to simply recompute all the `VersionEdit::remaining_entries_`s here? I find the fixup logic kind of hard to follow.

Reply:
This approach performs recomputation only when some column family has been dropped. Are you suggesting that we recompute this information anyway even if no column family is dropped?

Reply:
If it simplifies the code and doesn't increase the complexity, I'd vote for it.

Reply:
After some investigation, I still prefer the current approach because recomputing does not seem to simplify the code.

Reply:
Sounds good, thanks for looking into it!